### PR TITLE
feat: rewire xcm

### PR DESCRIPTION
**Pull request by alecdwm**
Switches out the underlying lib for [/transport/xcm](<https://app.talisman.xyz/transport/xcm>) to use `@galacticcouncil/xcm-sdk`.

### CODE REVIEW

File: apps/portal/src/components/widgets/xcm/api/utils/toPreciseDecimals.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { AssetAmount } from '@galacticcouncil/xcm-core'
+import BigNumber from 'bignumber.js'
+
+export const toPreciseDecimals = (amount?: AssetAmount): string | undefined =>
+  amount
+    ? BigNumber(String(amount.amount ?? 0n))
+        .times(BigNumber(10).pow(-1 * (amount.decimals ?? 0)))
+        .toString()
+    : undefined
```
Comment (chrisling-dev) [inline apps/portal/src/components/widgets/xcm/api/utils/toPreciseDecimals.ts:9]: seems like maybe `formatUnits` from viem can replace this

```suggestion
import { AssetAmount } from '@galacticcouncil/xcm-core'
import { formatUnits } from 'viem'

export const toPreciseDecimals = (amount?: AssetAmount): string | undefined =>
  amount ? formatUnits(amount.amount ?? 0n, amount.decimals ?? 1) : undefined
```

File: apps/portal/src/components/widgets/xcm/api/atoms/extrinsicAtom.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    //   'error.notValidAddr': 'The address is incorrect. Please review it and try again.',
+    //   'error.balance': 'Your transfer is bigger than your balance.',
+    //   'error.maxAmount': `The maximum transferable amount is ${amount} ${asset}.`,
+    //   'error.minAmount': `The minimum transferable amount is ${amount} ${asset}.`,
+
+    //   'error.fee.insufficientBalance': `You need to have ${amount} ${symbol} on ${chain} for fees`,
+    //   'error.destFee.insufficientBalance': `You need to have at least ${amount} ${symbol} on ${chain}`,
+    //   'error.asset.frozen': `Your account on ${chain} has frozen balance for ${symbol}`,
+    //   'error.account.insufficientDeposit': `You need to have ${amount} ${symbol} on ${chain} for existential deposit`,
+    // }
```
Comment (chrisling-dev) [inline apps/portal/src/components/widgets/xcm/api/atoms/extrinsicAtom.ts:61]: should this be removed or uncommented?

File: apps/portal/src/components/widgets/xcm/api/atoms/extrinsicAtom.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    const minWithRelay = min.copyWith({ amount: destChain.isEvmChain() ? min.amount * 2n : min.amount })
+
+    const amountBn = big.toBigInt(amount, balance.decimals)
+    const maxBn = big.toBigInt(max.amount, max.decimals)
+    const minBn = big.toBigInt(minWithRelay.amount, min.decimals)
+
+    if (balance.amount === 0n) {
+      throw new Error('Your transfer is bigger than your balance.')
+    }
```
Comment (chrisling-dev) [inline apps/portal/src/components/widgets/xcm/api/atoms/extrinsicAtom.ts:27]: Looking at the error message, should this be `balance.amount < amountBn`

**Review by chrisling-dev — APPROVED** [review]
Not familiar with xcm enough to provide any useful review, but did click around the feature to try things out and LGTM!

**Review by UrbanWill — APPROVED** [review]

**Pull request merged**